tests/unit/synapseclient/unit_test_client.py

```diff
@@ -1157,121 +1157,146 @@ def expected_request_json(token) -> str:
+                self.syn._check_entity_restrictions,
+                bundle,
+                entity,
+                True,
+            )
+        mocked_warn.assert_not_called()
 
+    def test_check_entity_restrictions_unmet_restriction_entity_project_with_download_file_is_true(
+        self,
+    ) -> None:
+        with patch("logging.Logger.warning") as mocked_warn:
+            bundle = {
+                "entity": {
+                    "id": "syn123",
+                    "name": "anonymous",
+                    "concreteType": "org.sagebionetworks.repo.model.FileEntity",
+                    "parentId": "syn12345",
+                },
+                "entityType": "project",
+                "restrictionInformation": {"hasUnmetAccessRequirement": True},
+            }
+            entity = "syn123"
+            self.syn._check_entity_restrictions(bundle, entity, True)
+        mocked_warn.assert_called_with(
+            "\nThis entity has access restrictions. Please visit the web page for this entity "
+            '(syn.onweb("syn123")). Look for the "Access" label and the lock icon underneath '
+            'the file name. Click "Request Access", and then review and fulfill the file '
+            "download requirement(s).\n"
+        )
 
-def test_check_entity_restrictions_unmet_restriction_entity_folder_with_download_file_is_true(
-    syn: Synapse,
-) -> None:
-    with patch("warnings.warn") as mocked_warn:
-        bundle = {
-            "entity": {
-                "id": "syn123",
-                "name": "anonymous",
-                "concreteType": "org.sagebionetworks.repo.model.FileEntity",
-                "parentId": "syn12345",
-            },
-            "entityType": "folder",
-            "restrictionInformation": {"hasUnmetAccessRequirement": True},
-        }
-        entity = "syn123"
-        syn._check_entity_restrictions(bundle, entity, True)
-    mocked_warn.assert_called_with(
-        "\nThis entity has access restrictions. Please visit the web page for this entity "
-        '(syn.onweb("syn123")). Look for the "Access" label and the lock icon underneath '
-        'the file name. Click "Request Access", and then review and fulfill the file '
-        "download requirement(s).\n"
-    )
+    def test_check_entity_restrictions_unmet_restriction_entity_folder_with_download_file_is_true_and_no_token(
+        self,
+    ) -> None:
+        with patch("logging.Logger.warning") as mocked_warn:
+            bundle = {
+                "entity": {
+                    "id": "syn123",
+                    "name": "anonymous",
+                    "concreteType": "org.sagebionetworks.repo.model.FileEntity",
+                    "parentId": "syn12345",
+                },
+                "entityType": "folder",
+                "restrictionInformation": {"hasUnmetAccessRequirement": True},
+            }
+            entity = "syn123"
+            self.syn.credentials = SynapseAuthTokenCredentials(token="")
```

Comment:
Would be worthwhile to have a test where login wasn't called at all, or syn.credentials is None